applications/CompressiblePotentialFlowApplication/custom_processes/define_embedded_wake_process.cpp

```diff
@@ -37,112 +36,93 @@ void DefineEmbeddedWakeProcess::Execute()
+                    r_geometry[i].SetValue(KUTTA, true);
+                    r_geometry[i].SetValue(WAKE_DISTANCE, nodal_distances_to_wake(i));
+                    r_geometry[i].UnSetLock();
                 }
-                it_elem->Set(ACTIVE, false);
             }
             else{
-                it_elem->SetValue(WAKE, true);
-                auto r_geometry = it_elem->GetGeometry();
-                for (unsigned int i = 0; i < it_elem->GetGeometry().size(); i++) {
+                auto& r_geometry = rElem.GetGeometry();
+                for (unsigned int i = 0; i < rElem.GetGeometry().size(); i++) {
                     r_geometry[i].SetLock();
                     r_geometry[i].SetValue(WAKE_DISTANCE, nodal_distances_to_wake(i));
+                    r_geometry[i].SetValue(WAKE, true);
                     r_geometry[i].UnSetLock();
                 }
             }
         }
-    }
-    deactivated_model_part.AddElements(deactivated_elements_id_list);
-}
+    });
 
-void DefineEmbeddedWakeProcess::ComputeTrailingEdgeNode(){
-
-    double max_distance = 0.0;
-    ModelPart& deactivated_model_part = mrModelPart.GetSubModelPart("deactivated_model_part");
-    Element::Pointer p_max_elem;
-
-    auto wake_origin = mrModelPart.GetProcessInfo()[WAKE_ORIGIN];
+    KRATOS_CATCH("");
 
-    // Find furthest deactivated element to the wake origin
-    for (int i = 0; i < static_cast<int>(deactivated_model_part.Elements().size()); i++) {
-        ModelPart::ElementIterator it_elem = deactivated_model_part.ElementsBegin() + i;
+}
 
-        BoundedVector<double,2> distance_vector;
-        distance_vector[0] = wake_origin[0] - it_elem->GetGeometry().Center().X();
-        distance_vector[1] = wake_origin[1] - it_elem->GetGeometry().Center().Y();
-        double norm = norm_2(distance_vector);
-        if(norm>max_distance){
-            max_distance = norm;
-            p_max_elem = mrModelPart.pGetElement(it_elem->Id());
-        }
-    }
+ModelPart::NodeType::Pointer DefineEmbeddedWakeProcess::pGetTrailingEdgeNode(){
 
-    // Mark nodes of the furthest deactivated element and store its neighbour elements
-    for (unsigned int i_node= 0; i_node < p_max_elem->GetGeometry().size(); i_node++) {
-        p_max_elem->GetGeometry()[i_node].SetValue(TRAILING_EDGE,true);
-        const GlobalPointersVector<Element>& r_node_elem_candidates = p_max_elem -> GetGeometry()[i_node].GetValue(NEIGHBOUR_ELEMENTS);
-        for (std::size_t j = 0; j < r_node_elem_candidates.size(); j++) {
-            mKuttaWakeElementCandidates.push_back(r_node_elem_candidates(j));
+    for (auto& r_node : mrModelPart.Nodes()) {
+        bool is_positive = r_node.GetValue(WAKE_DISTANCE) > 0.0;
+        bool is_wake = r_node.GetValue(WAKE);
+        bool is_kutta = r_node.GetValue(KUTTA);
+        if (is_positive && is_wake && is_kutta) {
+            r_node.SetValue(TRAILING_EDGE, true);
+            return &r_node;
         }
     }
```

Comment:
@philbucher What kind of reduction could be applied here to parallelize this? In order to find the node that meets all the conditions

Reply:
the node or the nodes?

Reply:
One node in this case (2D)

For 3D (not contemplated in this PR) there will be more than one, so maybe it is better to think already about filling a vector in parallel  `vector.push_back(&r_node)`  and return a list of pointers.

Reply:
one node is not possible as breaking from OpenMP loops is not possible

However multiple nodes is possible, [this](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/reduction_utilities.h#L188-L214) should work for you :)

Reply:
Ok thanks! I will leave it as it is for now.